### PR TITLE
resolved.conf via pillar

### PR DESCRIPTION
**Pull request by aboe76**
added option to configure resolved.conf via pillar using https://docs.saltstack.com/en/latest/ref/states/all/salt.states.ini_manage.html#salt.states.ini_manage.options_present

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -48,7 +48,11 @@ This state installs systemd-networkd profile files from pillar see pillar.exampl
 
 ``systemd.resolved``
 --------------------
-This state installs systemd-resolved and adds the timesyncd.conf file per os_family/minion_id
+This state installs systemd-resolved  and recursively adds file per os_family/minion_id
```
Comment (myii) [inline README.rst]: * Double space before `and`.
* `file` => `files`.

File: README.rst [hunk 2]
```diff
@@ -48,7 +48,11 @@ This state installs systemd-networkd profile files from pillar see pillar.exampl
...
+
+``systemd.resolved.config``
+--------------------
+This state installs systemd-resolved and adds the resolved.conf from pillar see pillar.example
```
Comment (myii) [inline README.rst]: * Either brackets: `(see pillar.example)`.
* Or punctuation: `, see pillar.example`.

File: systemd/resolved/config.sls [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{%- from "systemd/map.jinja" import systemd with context -%}
```
Comment (myii) [inline systemd/resolved/config.sls]: Perhaps leave off the extra whitespace control so only `{%- ... %}` (removing the second `-`).

File: systemd/resolved/config.sls [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{%- set resolved = systemd.get('resolved', {}) %}
+{%- set config = resolved.get('config', {}) %}
+
+resolved:
+  {% if resolved.pkg %}
```
Comment (myii) [inline systemd/resolved/config.sls]: Not important: `{%-`.

File: systemd/resolved/config.sls [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  pkg.installed:
+    - name: {{ resolved.pkg }}
+  {% endif %}
```
Comment (myii) [inline systemd/resolved/config.sls]: Not important: `{%-`.

**Comment by myii** [comment]
A quick eyeball review for now.  This looks good, I'll try to test it out within the next day, to leave further feedback.  Thanks.

**Comment by aboe76** [comment]
@myii thanks for looking over the PR, fixed the small issues.

**Comment by myii** [comment]
@aboe76 No problem, I'm testing out the PR now, so I'll be giving you some feedback shortly.

File: systemd/defaults.yaml [hunk 6]
```diff
@@ -20,3 +20,16 @@ systemd:
...
+    config:
+      DNS: '""'
+      FallbackDNS: '8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844'
+      Domains: '""'
+      LLMNR: 'yes'
+      MulticastDNS: 'yes'
+      DNSSEC: 'allow-downgrade'
+      DNSOverTLS: 'no'
+      Cache: 'yes'
+      DNSStubListener: 'udp'
```
Comment (myii) [inline systemd/defaults.yaml]: Use `yes` as the default?  See following reference.

https://www.freedesktop.org/software/systemd/man/resolved.conf.html#DNSStubListener=

> ... If "yes" (the default), the stub listens for both UDP and TCP requests ...

File: pillar.example [hunk 7]
```diff
@@ -44,6 +44,20 @@ systemd:
...
+      DNS: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001
+      FallbackDNS: 8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
+      Domains: '""'
+      LLMNR: 'True'
+      MulticastDNS: 'True'
+      DNSSEC: 'allow-downgrade'
+      DNSOverTLS: 'False'
+      Cache: 'True'
+      DNSStubListener: 'udp'
+      ReadEtchHosts: 'True'
```
Comment (myii) [inline pillar.example]: `ReadEtchHosts` => `ReadEtcHosts`.

File: pillar.example [hunk 8]
```diff
@@ -44,6 +44,20 @@ systemd:
...
+    config:
+      DNS: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001
+      FallbackDNS: 8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
+      Domains: '""'
+      LLMNR: 'True'
+      MulticastDNS: 'True'
+      DNSSEC: 'allow-downgrade'
+      DNSOverTLS: 'False'
+      Cache: 'True'
+      DNSStubListener: 'udp'
```
Comment (myii) [inline pillar.example]: Likewise, [this comment](https://github.com/saltstack-formulas/systemd-formula/pull/28/files?file-filters%5B%5D=.example&file-filters%5B%5D=.sls&file-filters%5B%5D=.yaml#r268825377).

**Comment by myii** [comment]
@aboe76 Not sure about the inconsistencies between `defaults.yaml` (represented below with `-`) and `pillar.example` (represented below with `+`).  Unless the intention is to show the various ways options can be set.

```yaml
+      DNS: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001
-      FallbackDNS: '8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844'
+      FallbackDNS: 8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
```

* `FallbackDNS` should lose the quotes?

```yaml
-      LLMNR: 'yes'
+      LLMNR: 'True'

-      MulticastDNS: 'yes'
+      MulticastDNS: 'True'

-      DNSOverTLS: 'no'
+      DNSOverTLS: 'False'

-      Cache: 'yes'
+      Cache: 'True'
```

* Better to stick to either `yes/no` or `True/False`?

**Comment by myii** [comment]
@aboe76 The `yes/no` gets converted to `True/False` at my end.  To preserve that, it works using `| yaml` like so:

```salt
    - sections:
        Resolve:
          DNS: {{ config.DNS | yaml }}
          FallbackDNS: {{ config.FallbackDNS | yaml }}
          Domains: {{ config.Domains | yaml }}
          LLMNR: {{ config.LLMNR | yaml }}
          MulticastDNS: {{ config.MulticastDNS | yaml }}
          DNSSEC: {{ config.DNSSEC | yaml }}
          DNSOverTLS: {{ config.DNSOverTLS | yaml }}
          Cache: {{ config.Cache | yaml }}
          DNSStubListener: {{ config.DNSStubListener | yaml }}
          ReadEtcHosts: {{ config.ReadEtcHosts | yaml }}
```

File: systemd/defaults.yaml (hunk 6 above)
Comment (aboe76) [reply systemd/defaults.yaml]: oops hihi will change it.

**Comment by aboe76** [comment]
@myii fixed the defaults and pillar.example with your suggestions.

File: pillar.example (hunk 7 above)
Comment (aboe76) [reply pillar.example]: oops fixed it

File: pillar.example [hunk 9]
```diff
@@ -44,6 +44,20 @@ systemd:
       Install:
         WantedBy: multi-user.target
 
+  ## resolved
+  resolved:
+    config:
+      DNS: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001
+      FallbackDNS: 8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
+      Domains: '""'
```
Comment (myii) [inline pillar.example]: After introducing `| yaml`: `'""'` => `''`.

File: systemd/defaults.yaml [hunk 10]
```diff
@@ -20,3 +20,16 @@ systemd:
     pkg: {}
     resolv_file: /etc/resolv.conf
     resolv_target: /usr/lib/systemd/resolv.conf
+    config:
+      DNS: '""'
```
Comment (myii) [inline systemd/defaults.yaml]: After introducing `| yaml`: `'""'` => `''`.

File: systemd/defaults.yaml [hunk 11]
```diff
@@ -20,3 +20,16 @@ systemd:
...
+      FallbackDNS: 8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844
+      Domains: '""'
```
Comment (myii) [inline systemd/defaults.yaml]: After introducing `| yaml`: `'""'` => `''`.

**Comment by myii** [comment]
@aboe76 Just the last thing I think!  No longer need the double-quotes inside the "empty" quotes.

File: systemd/defaults.yaml (hunk 11 above)
Comment (aboe76) [reply systemd/defaults.yaml]: done

File: systemd/defaults.yaml (hunk 10 above)
Comment (aboe76) [reply systemd/defaults.yaml]: done

File: pillar.example (hunk 9 above)
Comment (aboe76) [reply pillar.example]: done

**Comment by aboe76** [comment]
@myii will do something similar with timesync when this is merged, that way you can define everything by files or via pillar.

**Comment by myii** [comment]
@aanriot Thanks for the review.
@aboe76 Merged.

**Pull request merged**